### PR TITLE
MERL-1712: custom toolbar example

### LINKED ISSUE #1712: Create Custom Toolbar Example
## User Story (Required)

As a Faust.js user, I want to see an example of how to customize the toolbar so that I can see a succinct working example.

## Acceptance Criteria (Required)

* New custom toolbar example exists within the [faustjs repo](https://github.com/wpengine/faustjs/tree/canary/examples/next) (examples/next/custom-toolbar)
    * The custom toolbar example only contains the bare minimum for an accelerated understanding of how it works.
    * Example should only load the Toolbar’s styles (@faustwp/core/dist/css/toolbar.css)
* Link existing documentation to the new example to make it easier for users to make use of
* Add a readme to the example folder with links to proper documentation 

## Technical Details

* You’ll want to use a custom plugin in order to hook into the Faust toolbar filters - see [Customizing the Toolbar](https://faustjs.org/guide/how-to-customize-the-toolbar)

## Notes

TBD

### PR DESCRIPTION
**Pull request by matthewguywright**
## Description
Added an example for customizing the toolbar.

## Testing

1. Setup a faust project using the custom-toolbar example
2. Login using the button on the main frontpage template to authenticate
3. Toolbar should display with the custom elements showing from CustomToolbar.tsx in the /plugins folder.

## Screenshots
Logged Out:
Screenshot 2024-01-12 at 10 09 19 AM

Logged In:
Screenshot 2024-01-12 at 10 09 28 AM

Closes #1712

### CODE REVIEW

File: examples/next/custom-toolbar/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Welcome to the Faust Custom Toolbar Example.
+
+## Setup
+
+See the [Getting Started](https://faustjs.org/tutorial/get-started-with-faust) to setup your base Faust project. In the tutorial you will see in the Quick Start section's first step that it says `To create a Faust project, run`:
```
Comment (mindctrl) [inline examples/next/custom-toolbar/README.md:5]:
```suggestion
See the [Getting Started](https://faustjs.org/tutorial/get-started-with-faust) to set up your base Faust project. In the tutorial you will see in the Quick Start section's first step that it says `To create a Faust project, run`:
```

File: examples/next/custom-toolbar/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    -e https://github.com/wpengine/faustjs/tree/main \
+    --example-path examples/next/custom-toolbar \
+    --use-npm
+```
+
+After this one change you can continue the rest of the guide per the instructions.
+
+## Viewing the Toolbar
+
+To view the toolbar, make sure you have completed the Getting Started steps and then click `Login`. After you are logged in and authenticated you will see the toolbar at the top of the template.
```
Comment (mindctrl) [inline examples/next/custom-toolbar/README.md:27]:
```suggestion
To view the toolbar, make sure you have completed the Getting Started steps and then click `Log In`. After you are logged in and authenticated you will see the toolbar at the top of the template.
```

File: examples/next/custom-toolbar/wp-templates/front-page.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        menuItems={primaryMenu}
+      />
+      <Main>
+        <Container>
+          <Hero title={'Custom Toolbar Example'} />
+          <div className="text-center">
+            <p>
+              This page is utilizing the "front-page" WordPress template. To
+              authenticate and view the custom toolbar, click on the{' '}
+              <code>Login</code> button.
```
Comment (mindctrl) [inline examples/next/custom-toolbar/wp-templates/front-page.js:107]:
```suggestion
              <code>Log In</code> button.
```

File: examples/next/custom-toolbar/wp-templates/front-page.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            </p>
+            <code>wp-templates/front-page.js</code>
+          </div>
+          <div className="text-center">
+            <button onClick={loginHandler}>Login</button>
```
Comment (mindctrl) [inline examples/next/custom-toolbar/wp-templates/front-page.js:112]:
```suggestion
            <button onClick={loginHandler}>Log In</button>
```

File: examples/next/custom-toolbar/wp-templates/front-page.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+          <div className="text-center">
+            <p>
+              Welcome {data?.viewer?.name}! Look up! ^ Say hello to the
+              customized toolbar.
+            </p>
+            <p>
+              As long as you are authenticated you will see the toolbar. To
+              customize me further, go to <code>plugins/CustomPlugin.tsx</code>.
+              To logout and go back to the unauthenticated page, click on the{' '}
+              <code>Logout</code> button.
```
Comment (mindctrl) [inline examples/next/custom-toolbar/wp-templates/front-page.js:50]:
```suggestion
              To log out and go back to the unauthenticated page, click on the{' '}
              <code>Log Out</code> button.
```

File: examples/next/custom-toolbar/wp-templates/front-page.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+            </p>
+            <code>wp-templates/front-page.js</code>
+          </div>
+          <div className="text-center">
+            <button onClick={() => logout()}>Logout</button>
```
Comment (mindctrl) [inline examples/next/custom-toolbar/wp-templates/front-page.js:55]:
```suggestion
            <button onClick={() => logout()}>Log Out</button>
```

**Review by mindctrl — CHANGES_REQUESTED** [review]
Love the example being documentation. Nice work. Made some grammar suggestions.

**Comment by theodesp** [comment]
@matthewguywright Do we really need a full blown example project here?

- Developers might get confused about where to find the new toolbar code.
- We have another copy of the site which is code duplication.

I suggest you remove all the extra parts and keep it very simple and with not duplicated components.

**Comment by matthewguywright** [comment]
Moving to a more bare bones setup.

**Pull request closed without merging**